pyuvdata/uvdata/mir_parser.py

```diff
@@ -923,36 +1106,640 @@ def _apply_tsys(self, jypk=130.0):
+            a corresponding key in the relevant data attribute, indicating that there
+            are records requested that are not loaded into memory.
+        """
+        select_vis = (self.vis_data is not None) if (select_vis is None) else select_vis
+        select_raw = (self.raw_data is not None) if (select_raw is None) else select_raw
+        select_auto = (
+            (self.auto_data is not None) if (select_auto is None) else select_auto
+        )
+
+        try:
+            # Check that vis_data has all entries we need for processing the data
+            if select_vis:
+                vis_data = {
+                    sphid: self.vis_data[sphid] for sphid in self.sp_data["sphid"]
+                }
+
+            # Now check raw_data
+            if select_raw:
+                raw_data = {
+                    sphid: self.raw_data[sphid] for sphid in self.sp_data["sphid"]
+                }
+
+            # Now check auto_data
+            if select_auto:
+                auto_data = {
+                    achid: self.auto_data[achid] for achid in self.ac_data["achid"]
+                }
+        except (TypeError, KeyError):
+            raise MirMetaError(
+                "Missing spectral records in data attributes. Run load_data instead."
+            )
+
+        # At this point, we can actually plug our values in, since we know that the
+        # operation above succeeded.
+        if select_vis and self.vis_data is not None:
+            self.vis_data = vis_data
+        if select_raw and self.raw_data is not None:
+            self.raw_data = raw_data
+        if select_auto and self.auto_data is not None:
+            self.auto_data = auto_data
+
+    def load_data(
+        self,
+        load_vis=None,
+        load_raw=None,
+        load_auto=None,
+        apply_tsys=True,
+        allow_downselect=None,
+        allow_conversion=None,
+        use_mmap=True,
+        read_only=False,
+    ):
+        """
+        Load visibility data into MirParser class.
+
+        Method for loading the visibility data into the object. Can either load the
+        compressed "raw data" (which is about half the size in memory, but carries
+        overheads in various function calls _except_ for writing to disk) or the
+        uncompressed "vis data" (floating point values).
+
+        Note that this function will only load either  vis data or raw data, not
+        both. If told to load both, then the method will default to loading only the
+        vis data.
+
+        Parameters
+        ----------
+        load_vis : bool
+            Load the visibility data (floats) into object. Default is True if
+            `load_raw` is unset or otherwise set to False.
+        load_raw : bool
+            Load the raw visibility data (ints) into object. Default is False, unless
+            `load_vis` is set to False, in which case it defaults to True.
```

Comment:
Since these are mutually exclusive, it might make sense to combine them into a single parameter.

Reply:
Changed this in a2f46b4b.